### PR TITLE
fix: subpackage changes not autoreloaded

### PR DESCRIPTION
**Pull request by iisakkirotko**
Fixes issue #281.

### CODE REVIEW

File: solara/server/app.py [hunk 1]
```diff
@@ -79,6 +79,11 @@ def __init__(self, name, default_app_name="Page"):
 
         self._first_execute_app = app
         reload.reloader.root_path = self.directory
+        if self.type == AppType.MODULE:
+            package_name = self.name.split(".")[0]
+            mod = importlib.import_module(package_name)
+            package_root_path = Path(mod.__file__).parent
+            reload.reloader.root_path = package_root_path
```
Comment (maartenbreddels) [inline solara/server/app.py:86]: I think this needs to be conditional (looking at the mypy error) when mod.__file__ is None, we just skip this I think.

Comment (iisakkirotko) [reply solara/server/app.py]: Made the change. Shouldn't worry too much about the ~30 errors from Python 3.9?

**Comment by maartenbreddels** [comment]
Also fixes #242

**Comment by maartenbreddels** [comment]
Thanks 👍

**Pull request merged**